packages/utilities/exposure-scanning/scan/index.js

```diff
@@ -0,0 +1,77 @@
+const replacementStr = '*';
+
+const scanText = (text, regex) => {
+  let foundTextMatch = false;
+  const scannedText = text.replace(regex, replacementStr);
+
+  if (scannedText !== text) {
+    foundTextMatch = true;
+  }
+
+  return { body: scannedText, foundTextMatch };
+};
+
+const scanComments = (comments, regex) => {
+  let foundCommentMatch = false;
+  const scannedComments = [];
+
+  comments.forEach(({ id, body }) => {
+    if (regex.test(body)) {
+      foundCommentMatch = true;
+      scannedComments.push({
+        id,
+        body: body.replace(regex, replacementStr),
+      });
+    }
+  });
+
+  return { comments: scannedComments, foundCommentMatch };
+};
+
+export const scanPr = (pr, regex) => {
+  let scannedPr = pr;
+  let foundMatch = false;
+
+  try {
+    const { body, foundTextMatch } = scanText(pr.body, regex);
+    const { comments, foundCommentMatch } = scanComments(pr.comments, regex);
+    const {
+      comments: reviewComments,
+      foundCommentMatch: foundReviewCommentsMatch,
+    } = scanComments(pr.reviewComments, regex);
+
+    scannedPr = {
+      body,
+      comments,
+      reviewComments,
+    };
+
+    foundMatch =
+      foundTextMatch || foundCommentMatch || foundReviewCommentsMatch;
+  } catch (error) {
+    throw new Error(`Encountered an error when scanning.`);
```

Comment:
These errors will hide the cause of the original error as the stack trace will be lost; I'm not sure what catching the error achieves here as you throw another error? You normally catch an error and output to console for the user (e.g. `console.error`) and then either rethrow or return a suitable response to allow the application to continue as normal.